src/Microsoft.Framework.PackageManager/DependencyAnalyzer/DependencyFinder.cs

```diff
@@ -0,0 +1,218 @@
+        {
+            var results = new HashSet<string>();
+            var missing = new HashSet<string>();
+            var nonProjectLibs = _hostContext.LibraryManager.GetLibraries()
+                .Where(library => !string.Equals(LibraryTypeProject, library.Type, StringComparison.OrdinalIgnoreCase));
+            var libLookup = _hostContext.DependencyWalker.Libraries
+                .ToLookup(keySelector: desc => desc.Identity);
+
+            if (local)
+            {
+                foreach (var libraryInformation in nonProjectLibs)
+                {
+                    foreach (var loadableAssembly in libraryInformation.LoadableAssemblies)
+                    {
+                        IEnumerable<string> dependencies;
+                        dependencies = ResolveLocalDependency(loadableAssembly.Name, assemblyName => missing.Add(assemblyName));
+
+                        results.AddRange(dependencies);
+                    }
+                }
+            }
+            else
+            {
+                foreach (var libraryInformation in nonProjectLibs)
+                {
+                    results.AddRange(ResolveDependency(libraryInformation.Name, libLookup, assemblyName => missing.Add(assemblyName)));
+                }
+            }
+
+            unresolved = missing.ToList();
+
+            return results;
+        }
+
+        private ISet<string> ResolveDependency(string rootReference, ILookup<Library, LibraryDescription> libLookup, Action<string> onUnresovled)
+        {
+            var results = new HashSet<string>();
+            var rootLibrary = _hostContext.DependencyWalker.Libraries.FirstOrDefault(l => l.Identity.Name == rootReference)?.Identity;
+            if (rootLibrary == null)
+            {
+                onUnresovled(rootReference);
+            }
+
+            var stack = new Stack<Library>();
+            stack.Push(rootLibrary);
+
+            while (stack.Count > 0)
+            {
+                Library current = stack.Pop();
+
+                if (!results.Add(current.Name))
+                {
+                    continue;
+                }
+
+                var description = libLookup[current].FirstOrDefault();
+                if (description != null)
+                {
+                    foreach (var each in description.Dependencies.Select(dep => dep.Library))
+                    {
+                        stack.Push(each);
+                    }
+                }
+                else
+                {
+                    onUnresovled(current.Name);
+                }
+            }
+
+            return results;
+        }
+
+        private ISet<string> ResolveLocalDependency(string rootReference, Action<string> onUnresvoled)
+        {
+            var result = new HashSet<string>();
+            var stack = new Stack<string>();
+            var gacResolver = new GacDependencyResolver();
```

Comment:
Resolving local reference is based on the assembly information. If a given assembly is in GAC, there is no need to further walk down the tree. The resolver is the best way I can find to determine if the assembly is in GAC.